components/scrubber/scrubber.go

```diff
@@ -94,11 +106,22 @@ func (s *scrubberImpl) JSON(msg json.RawMessage) (json.RawMessage, error) {
 
 // KeyValue implements Scrubber
 func (s *scrubberImpl) KeyValue(key string, value string) (sanitisedValue string) {
-	f, ok := s.FieldIndex[strings.ToLower(key)]
-	if !ok {
-		return value
+	return s.GetSanitisatiser(key)(value)
+}
+
+// GetSanitisatiser implements
+func (s *scrubberImpl) GetSanitisatiser(key string) Sanitisatiser {
+	if f, ok := s.FieldIndex[key]; ok {
```

Comment:
(note) we don't expect many various fields in different cases in the same component, it allows to save allocation by avoiding conversions to lower case and boost performance to compensate for regex below

Reply:
We do have a few places where we spell things like `workspaceID` and `workspaceId`. We could just list them explicitly, exp. with the regexp matcher below.